### PR TITLE
Concerning #16 , #33

### PR DESCRIPTION
**Pull request by AlhasanIQ**
added initial manifest #16 
and a sketch for isCompatible #33

### CODE REVIEW

**Comment by fadeenk** [comment]
Are we going to add support for different triggers? See [this comment](https://github.com/Lamassau/PiTrol/issues/16#issuecomment-279278450)

**Comment by AlhasanIQ** [comment]
i did see it,
but i think that we should keep it simple until we get to Triggers.

File: server/app.js [hunk 1]
```diff
@@ -43,12 +48,46 @@ for (var i = 0; i < plugins.length; i++) {
...
+  } catch (e) {
+    if (e && e.msg) {
+      console.log(e.msg)
+    } else {
+      console.log(e)
+    }
+  }
+}
+
+function checkManifest (mainfest) {
```
Comment (fadeenk) [inline server/app.js]: Who is working on those checks??

Comment (AlhasanIQ) [inline server/app.js]: no one yet,
these are sketches that will always `return true` for now

**Comment by fadeenk** [comment]
@AlhasanIQ so when are we thinking of adding the different triggers? I was thinking we could add the system now then add the more advanced triggers later

**Comment by AlhasanIQ** [comment]
imo, we should prioritize finishing the plugin related things and get a couple working plugins + template before doing anything else.
porting the plugin to triggers wont be a big deal

**Comment by fadeenk** [comment]
@AlhasanIQ can you allow maintainers to modify your PR 😄 I have the logic to check for the manifest and compatibility but cant push it 😢

**Comment by AlhasanIQ** [comment]
how do i do that 😄

**Comment by fadeenk** [comment]
Not quite sure, typically I do it when I create the PR but [this](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) has some instructions that should work

**Comment by AlKass** [comment]
@AlhasanIQ check in your code and @fadeenk can pull the master branch, add his changes, and submit a pull request. that's about the easiest way you can go about doing what you're trying to do

**Comment by fadeenk** [comment]
@AlKass the instructions that I posted are kind of straight forward if that doesnt work, @AlhasanIQ can just give me access to his fork's master and that would work too 😄

**Comment by AlhasanIQ** [comment]
the thingy on the link you provided is already checked,

**Comment by fadeenk** [comment]
Just pushed it was my fault :P I was pushing to my master instead of yours so it wouldnt let me

**Comment by AlhasanIQ** [comment]
did review the code,
clean and neat.

**Comment by AlhasanIQ** [comment]
merge now or wait for `checkPluginsDependencies()` ?

**Comment by fadeenk** [comment]
checkPluginsDependencies is ganna require alot more work since we dont really know how that will work yet. So I am ok with merging what we have and adding an issue for missing checkPluginDeps function as well as checking for Pi hardware

Also I just split the validation logic from the app.js make things cleaner

**Comment by AlKass** [comment]
why don't we let the plugin decide whether it's compatible with the version of OS and the hardware instead of us doing all the heavy lifting? We could let each plugin define an isCompatible function that does all the checks needed to be carried out for the plugin to work, and all we do is invoke the function and expect a JSON object back containing two fields: `status` and `message`. `status` is either `true` or `false`, and `message` is the error we return to the user if the plugin is incompatible.
This is not only practice but also allows for the tool to run on all different platforms

**Comment by fadeenk** [comment]
because people would do this
```javascript
function isCompatible(){
return true;
}
```
And the plugin could break everything.

**Pull request merged**